pkg/analyzer/commit_analyzer.go

```diff
@@ -35,11 +35,23 @@ func (da *DefaultCommitAnalyzer) analyzeSingleCommit(rawCommit *semrel.RawCommit
 	}
 	c.Type = strings.ToLower(found[0][1])
 	c.Scope = found[0][2]
-	c.Message = found[0][3]
+	breakingChange := found[0][3]
+	c.Message = found[0][4]
+
+	isMajorChange := breakingPattern.MatchString(rawCommit.RawMessage)
+	isMinorChange := c.Type == "feat"
+	isPatchChange := c.Type == "fix"
+
+	if len(breakingChange) > 0 {
+		isMajorChange = true
+		isMinorChange = false
```

Comment:
Can we remove the following:

```
isMinorChange = false
isPatchChange = false
```

to ensure consistency?

Reply:
Hi, if I do remove them, these tests fail:
```
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_feat!:_modified_login_endpoint
c.Change: major:true minor:true
    commit_analyzer_test.go:96: 
        	Error Trace:	commit_analyzer_test.go:96
        	Error:      	Should be true
        	Test:       	TestDefaultAnalyzer/AnalyzeCommitMessage:_feat!:_modified_login_endpoint
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_fix!:_fixed_a_typo
c.Change: major:true patch:true
    commit_analyzer_test.go:96: 
        	Error Trace:	commit_analyzer_test.go:96
        	Error:      	Should be true
        	Test:       	TestDefaultAnalyzer/AnalyzeCommitMessage:_fix!:_fixed_a_typo
```
...as you can see, the `c.Change` for the first test contains both `major:true` and `minor:true` and for the second `major:true` and `patch:true`. And if we bump major version, minor and patch should be reseted, as well as if we bump minor, patch should be reseted. That's why I added these variables in the first place. 🙂

But I'm open for making changes, if you want me to change these somehow differently.

Reply:
Just in case you're curious:
```sh
?   	github.com/go-semantic-release/commit-analyzer-cz/cmd/commit-analyzer-cz	[no test files]
=== RUN   TestDefaultAnalyzer
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_feat:_new_feature
c.Change: minor:true
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_feat(web):_new_feature
c.Change: minor:true
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_new_feature
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_chore:_break_BREAKING_CHANGE:_breaks_everything
c.Change: major:true
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_feat!:_modified_login_endpoint
c.Change: major:true minor:true
    commit_analyzer_test.go:96: 
        	Error Trace:	commit_analyzer_test.go:96
        	Error:      	Should be true
        	Test:       	TestDefaultAnalyzer/AnalyzeCommitMessage:_feat!:_modified_login_endpoint
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_fix!:_fixed_a_typo
c.Change: major:true patch:true
    commit_analyzer_test.go:96: 
        	Error Trace:	commit_analyzer_test.go:96
        	Error:      	Should be true
        	Test:       	TestDefaultAnalyzer/AnalyzeCommitMessage:_fix!:_fixed_a_typo
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_refactor(parser)!:_drop_support_for_Node_6__BREAKING_CHANGE:_refactor_to_use_JavaScript_features_not_available_in_Node_6.
c.Change: major:true
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_docs:_added_more_documentation
c.Change: 
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_chore:_moved_README.md_to_root
c.Change: 
--- FAIL: TestDefaultAnalyzer (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_feat:_new_feature (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_feat(web):_new_feature (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_new_feature (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_chore:_break_BREAKING_CHANGE:_breaks_everything (0.00s)
    --- FAIL: TestDefaultAnalyzer/AnalyzeCommitMessage:_feat!:_modified_login_endpoint (0.00s)
    --- FAIL: TestDefaultAnalyzer/AnalyzeCommitMessage:_fix!:_fixed_a_typo (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_refactor(parser)!:_drop_support_for_Node_6__BREAKING_CHANGE:_refactor_to_use_JavaScript_features_not_available_in_Node_6. (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_docs:_added_more_documentation (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_chore:_moved_README.md_to_root (0.00s)
FAIL
FAIL	github.com/go-semantic-release/commit-analyzer-cz/pkg/analyzer	0.247s
FAIL
```
I only removed the lines you asked for and added `fmt.Printf("c.Change: %+v\n", c.Change)` in `pkg/analyzer/commit_analyzer.go:58` right before the `return`, to see what values does it return.

If I only add the `Printf` there, the output will be:
```sh
?   	github.com/go-semantic-release/commit-analyzer-cz/cmd/commit-analyzer-cz	[no test files]
=== RUN   TestDefaultAnalyzer
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_feat:_new_feature
c.Change: minor:true
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_feat(web):_new_feature
c.Change: minor:true
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_new_feature
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_chore:_break_BREAKING_CHANGE:_breaks_everything
c.Change: major:true
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_feat!:_modified_login_endpoint
c.Change: major:true
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_fix!:_fixed_a_typo
c.Change: major:true
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_refactor(parser)!:_drop_support_for_Node_6__BREAKING_CHANGE:_refactor_to_use_JavaScript_features_not_available_in_Node_6.
c.Change: major:true
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_docs:_added_more_documentation
c.Change: 
=== RUN   TestDefaultAnalyzer/AnalyzeCommitMessage:_chore:_moved_README.md_to_root
c.Change: 
--- PASS: TestDefaultAnalyzer (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_feat:_new_feature (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_feat(web):_new_feature (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_new_feature (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_chore:_break_BREAKING_CHANGE:_breaks_everything (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_feat!:_modified_login_endpoint (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_fix!:_fixed_a_typo (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_refactor(parser)!:_drop_support_for_Node_6__BREAKING_CHANGE:_refactor_to_use_JavaScript_features_not_available_in_Node_6. (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_docs:_added_more_documentation (0.00s)
    --- PASS: TestDefaultAnalyzer/AnalyzeCommitMessage:_chore:_moved_README.md_to_root (0.00s)
PASS
ok  	github.com/go-semantic-release/commit-analyzer-cz/pkg/analyzer	0.123s
```

Reply:
Okay, I agree. Let's leave it how it is ☺️